bin/runFixedPop

```diff
@@ -256,42 +260,50 @@ if __name__ == '__main__':
         dat_store.append('idx', pd.DataFrame([idx]))
 
         # Filter the bcm and bpp arrays according to user specified filters
-        bcm = filter_bpp_bcm(bcm, bpp, filters)
-        
-        # Filter out systems that don't match final kstars supplied
-        bcm_save_filtered = bcm.loc[(bcm.kstar_1.isin(kstar1_range)) & (bcm.kstar_2.isin(kstar2_range))]
-        
+        if len(bcm_filter) > 1:
+            bcm_filter = bcm_filter.append(filter_bpp_bcm(bcm, bpp, filters, kstar1_range, kstar2_range))
+            initC_filter = initC_filter.append(initCond.loc[initCond.bin_num.isin(bcm_filter.bin_num)])
+            bpp_filter = bpp_filter.append(bpp.loc[bpp.bin_num.isin(bcm_filter.bin_num)])
+        else:
+            bcm_filter = filter_bpp_bcm(bcm, bpp, filters, kstar1_range, kstar2_range)
+            initC_filter = initCond.loc[initCond.bin_num.isin(bcm_filter.bin_num)]
+            bpp_filter = bpp.loc[bpp.bin_num.isin(bcm_filter.bin_num)]
+
         # Run the match on short period binaries, such that we get a better sampling
         # for the highest signal systems
-        if (len(bcm_save_filtered) > 0):
+        n_bin_state = []
+        for bin_state in filters['binary_state']:
+            n_bin_state.append(len(bcm_filter.loc[bcm_filter.bin_state == bin_state]))
+        if all(i >= 50 for i in n_bin_state):
             # Save the bcm dataframe
-            dat_store.append('bcm', bcm_save_filtered)
-            bcm_save = bcm_save.append(bcm_save_filtered)
+            dat_store.append('bcm', bcm_filter)
+            bcm_save = bcm_save.append(bcm_filter)
            
-            idxSave = bcm_save_filtered.bin_num
-            bpp_save_filtered = bpp[bpp.bin_num.isin(idxSave)]
-
             # Save the bpp dataframe
-            dat_store.append('bpp', bpp_save_filtered)
-            bpp_save = bpp_save.append(bpp_save_filtered)
-
-            # Now let's select out the initial binaries that produce our data
-            initC = initCond.loc[initCond.bin_num.isin(idxSave)]
+            dat_store.append('bpp', bpp_filter)
+            bpp_save = bpp_save.append(bpp_filter)
 
             # Save the initial binaries 
-            dat_store.append('initCond', initC)
+            dat_store.append('initCond', initC_filter)
 
             # perform the convergence
             match_all = Match.perform_convergence(args.convergence_params, filters['binary_state'],\
-                                                  convergence, bcm_save, bcm_save_filtered, bpp_save,\
+                                                  convergence, bcm_save, bcm_filter, bpp_save,\
                                                   kstar1_range, kstar2_range, log_file)
+ 
             if len(match_all) > 1:
+                match_all = np.reshape(match_all, (-1,len(filters['binary_state'])))
```

Comment:
I *think* based on my testing of only one bin_state we want this to be `len(args.convergence_params)`

Reply:
You are right: This is fixed [here](https://github.com/COSMIC-PopSynth/COSMIC/pull/107/commits/52a7e8400424f00caf7438fe04ede2906420e80a#diff-30a06cad5f4b8c2bdc7fcbbad5e6305aR295).